macros/commands/dump_table.sql

```diff
@@ -1,14 +1,24 @@
-{% macro dump_table(model_unique_id, output_path, since=none, days_back=7) %}
+{% macro dump_table(model_unique_id, output_path, exclude_deprecated_columns=true, since=none, days_back=7) %}
     {% set node = graph.nodes[model_unique_id] %}
-    {% set relation_exists = adapter.get_relation(database=node.database, schema=node.schema, identifier=node.alias) %}
-    {% if not relation_exists %}
+    {% set relation = adapter.get_relation(database=node.database, schema=node.schema, identifier=node.alias) %}
+    {% if relation is none %}
         {% do print("Relation '{}' does not exist.".format(node.relation_name)) %}
         {% do return([]) %}
     {% endif %}
 
+    {% set column_names = adapter.get_columns_in_relation(relation) | map(attribute="name") | list %}
+    {% if exclude_deprecated_columns %}
+        {% set deprecated_column_names = node.columns.values()| selectattr("deprecated") | map(attribute="name") | list %}
+        {% for deprecated_column_name in deprecated_column_names %}
+            {% if deprecated_column_name in column_names %}
+                {% do column_names.remove(deprecated_column_name) %}
```

Comment:
instead of adding columns to the node and marking as deprecated, won't it make more sense to intersect the columns from the info schema with the columns from the node? And instead of deprecating - just deleting the column from the yaml?

Reply:
Guessing the answer is equivalent to the following question: Do we ever run this macro with `exclude_deprecated_columns`=False ?

Reply:
I totally agree regarding the first point, the problem is that at the moment we don't maintain the columns list (some models don't even have it). It'd make sense to do it if we had a test that compares the table's state against the columns definition, but this will take some time and this is a blocker for a user.

Regarding the second question, no.

Reply:
Seems OK in this case to move forward